admin/class-yoast-notification-center.php

```diff
@@ -25,12 +25,18 @@ class Yoast_Notification_Center {
 	/** @var array Notifications that were resolved this execution */
 	private $resolved = 0;
 
+	/** @var array Internal storage for transaction before notifications have been retrieved from storage. */
+	private $queued_transactions = array();
+
+	/** @var bool Internal flag for whether notifications have been retrieved from storage. */
+	private $notifications_retrieved = false;
+
 	/**
 	 * Construct
 	 */
 	private function __construct() {
 
-		$this->retrieve_notifications_from_storage();
+		add_action( 'init', array( $this, 'setup_current_notifications' ), 1 );
```

Comment:
Have you considered using a negative value for the priority?

Reply:
We could do that, but I'm not sure what benefit it would give. I would only recommend negative priorities if we absolutely need to ensure nothing happens on the same hook before, which I don't think is the case here.